--- ./nuttx/drivers/audio/Make.defs	2019-04-23 10:11:11.000000000 -0700
+++ ./nuttx/drivers/audio/Make.defs	2019-04-29 11:37:16.214021005 -0700
@@ -65,6 +65,10 @@
 endif
 endif
 
+ifeq ($(CONFIG_AUDIO_ADAU1361),y)
+CSRCS += adau1361.c
+endif
+
 ifeq ($(CONFIG_AUDIO_WM8776),y)
 CSRCS += wm8776.c
 endif
--- ./nuttx/drivers/audio/Kconfig	2019-04-29 11:31:25.363037866 -0700
+++ ./nuttx/drivers/audio/Kconfig	2019-04-29 11:36:06.889484291 -0700
@@ -62,6 +62,13 @@
 endif # AUDIO_DRIVER_SPECIFIC_BUFFERS
 endif # AUDIO_VS1053
 
+config AUDIO_ADAU1361
+	bool "ADAU1361 stereo codec"
+	default n
+	depends on AUDIO
+	---help---
+		Select to enable support for the Analog Devices ADAU1361 Stereo CODEC.
+
 config AUDIO_CS43L22
 	bool "CS43L22 audio chip"
 	default n
